### PR TITLE
Append `aws-chunked` to `Content-Encoding` header if present

### PR DESCRIPTION
**Pull request by stealthycoin**
sourced from https://github.com/boto/botocore/pull/2859

### CODE REVIEW

**Comment by dlm6693** [comment]
@stealthycoin I think you need to add this commit as well: https://github.com/boto/botocore/commit/8d2141e1a94bc1477d032457dec5de91d632d95d

File: .changes/next-release/bugfix-s3-10151.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "type": "bugfix",
+  "category": "``s3``",
+  "description": "boto3 no longer overwrites user supplied `Content-Encoding` with `aws-chunked` when user also supplies `ChecksumAlgorithm`."
```
Comment (dlm6693) [inline .changes/next-release/bugfix-s3-10151.json:4]: We should change the description to be for `aws-cli` not `boto3`

**Review by dlm6693 — CHANGES_REQUESTED** [review]
lgtm! Just update the changelog and we can ship it.

**Review by kyleknap — APPROVED** [review]
:ship:

**Review by dlm6693 — APPROVED** [review]

**Pull request merged**